create-presigned-download-url/src/main/java/no/unit/nva/download/publication/file/CreatePresignedDownloadUrlHandler.java

```diff
@@ -86,20 +86,27 @@ private boolean findByIdentifier(UUID fileIdentifier, File element) {
         return fileIdentifier.equals(element.getIdentifier());
     }
 
-    private boolean isFindable(String user,
-                               File file,
-                               Publication publication,
-                               RequestInfo requestInfo) {
-        return publication.getResourceOwner().getOwner().getValue().equals(user)
-                || requestInfo.userIsAuthorized(EDIT_OWN_INSTITUTION_RESOURCES.toString())
-                || PublicationStatus.PUBLISHED.equals(publication.getStatus()) && file.isVisibleForNonOwner();
+    private boolean hasReadAccess(File file, Username owner, PublicationStatus status, RequestInfo requestInfo) {
+
+        var isEmbargoReader = requestInfo.userIsAuthorized(PUBLISH_THESIS_EMBARGO_READ.toString());
```

Comment:
I would expect the PUBLISH_THESIS_EMBARGO_READ access right to only apply when we are dealing with a publication where `entityDescription.getReference().getPublicationContext().getType()` is equal to `Degree`. If not, it should behave as before. Correct?

Reply:
Shall we supply embargoed files as long as they are not a thesis?